File: java/src/test/java/com/incidentcommander/ModelsTest.java
package com.incidentcommander;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

import java.util.List;
import org.junit.jupiter.api.Test;

class ModelsTest {
    @Test
    void allServicesStableOrder() {
        assertEquals(List.of("api", "payment", "database", "cache", "queue"), Services.ALL);
    }

    @Test
    void stateConstantsAreReadableStrings() {
        assertEquals("NEW", States.NEW);
        assertEquals("WAITING_FOR_APPROVAL", States.WAITING_FOR_APPROVAL);
        assertEquals("low", Safety.LOW);
        assertEquals("critical", Severity.CRITICAL);
    }

    private static Hypothesis hyp(String cause, double confidence) {
        Hypothesis h = new Hypothesis();
        h.cause = cause;
        h.confidence = confidence;
        return h;
    }

    @Test
    void diagnosisTopReturnsHighestConfidence() {
        Diagnosis d = new Diagnosis();
        d.hypotheses = List.of(hyp("a", 0.3), hyp("b", 0.9), hyp("c", 0.5));
        assertEquals("b", d.top().cause);
    }

    @Test
    void diagnosisTopTiesKeepFirst() {
        Diagnosis d = new Diagnosis();
        d.hypotheses = List.of(hyp("first", 0.8), hyp("second", 0.8));
        assertEquals("first", d.top().cause);
    }

    @Test
    void diagnosisTopEmptyIsNull() {
        assertNull(new Diagnosis().top());
    }

    @Test
    void diagnosisTopHandlesZeroConfidence() {
        Diagnosis d = new Diagnosis();
        d.hypotheses = List.of(hyp("only", 0.0));
        assertEquals("only", d.top().cause);
    }
}

File: rust/tests/models_test.rs
use incident_commander::models::{
    Diagnosis, Hypothesis, ALL_SERVICES, SAFETY_LOW, SEVERITY_CRITICAL, STATE_NEW,
    STATE_WAITING_FOR_APPROVAL,
};

#[test]
fn all_services_stable_order() {
    assert_eq!(
        ALL_SERVICES,
        ["api", "payment", "database", "cache", "queue"]
    );
}

#[test]
fn state_constants_are_readable_strings() {
    assert_eq!(STATE_NEW, "NEW");
    assert_eq!(STATE_WAITING_FOR_APPROVAL, "WAITING_FOR_APPROVAL");
    assert_eq!(SAFETY_LOW, "low");
    assert_eq!(SEVERITY_CRITICAL, "critical");
}

fn hyp(cause: &str, confidence: f64) -> Hypothesis {
    Hypothesis {
        cause: cause.to_string(),
        confidence,
        ..Default::default()
    }
}

#[test]
fn diagnosis_top_returns_highest_confidence() {
    let d = Diagnosis {
        hypotheses: vec![hyp("a", 0.3), hyp("b", 0.9), hyp("c", 0.5)],
        ..Default::default()
    };
    assert_eq!(d.top().unwrap().cause, "b");
}

#[test]
fn diagnosis_top_ties_keep_first() {
    let d = Diagnosis {
        hypotheses: vec![hyp("first", 0.8), hyp("second", 0.8)],
        ..Default::default()
    };
    assert_eq!(d.top().unwrap().cause, "first");
}

#[test]
fn diagnosis_top_empty_is_none() {
    assert!(Diagnosis::default().top().is_none());
}

#[test]
fn diagnosis_top_handles_zero_confidence() {
    let d = Diagnosis {
        hypotheses: vec![hyp("only", 0.0)],
        ..Default::default()
    };
    assert_eq!(d.top().unwrap().cause, "only");
}
